### PR TITLE
[VEGA-4292] fix: dropdown opened when scrolling

### PR DESCRIPTION
**Pull request by apequa-alan**
## Problem statement

https://jira.vegaspace.tk/browse/VEGA-4292

## Solution details

Добавлен фикс на наслаивание выпадающего списка 'категории'

## Type

- [ ] Feature
- [x] Bug Fix
- [ ] Refactoring
- [ ] Test
- [ ] Other

## Associated issues

_{Jira issues or PRs related to this one}_

## Quality control

- [x] PR: Based on a correct branch
- [x] PR: Head branch is rebased on the base branch
- [ ] PR: Assignee chosen and necessary labels are set
- [ ] PR: Current form is filled out (where it makes sense)
- [ ] PR: Diff checked for irrelevant changes
- [x] PR: Commit messages follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specs
- [x] PR: Checked for spelling, grammar and typos
- [x] JS: There's no errors/warnings in the browser dev console
- [ ] Layout: Tested with various content amounts
- [ ] Docs: All the important changes and features are described
- [ ] Tests: Configuration files are checked at test repositories
- [ ] Tests: New unit tests developed
- [ ] Tests: New E2E tests developed
- [x] Tests: Existing unit tests passed successfully
image
- [x] Tests: Existing E2E tests passed successfully
image
- [ ] Tests: Manually tested on personal instance

## Notable statements
- [ ] Affects configuration files
- [ ] Brings new packages or updates existing
- [ ] Opened follow-up tasks to resolve

### CODE REVIEW

File: src/components/ExcelTable/ExcelTable.tsx [hunk 1]
```diff
@@ -401,6 +401,34 @@ export const ExcelTable: React.FC<ExcelTableProps> = ({
...
+        }
+      }
+    };
+    const exelTable = gridRef.current?.element;
+    exelTable?.addEventListener('scroll', scrollHandler);
+
+    return () => {
+      exelTable?.removeEventListener('scroll', scrollHandler);
+    };
+  });
```
[the remark this answers is not in the document]
Comment (apequa-alan) [reply src/components/ExcelTable/ExcelTable.tsx:430]: > Может вынести логику в хэлперы или т.п?
> 
> ```
> useEffect( () => {
>   const exelTable = gridRef.current.element;
>   exelTable.addEventListener('scroll', closeDropdownOnScroll);
>   return () =>  exelTable.removeEventListener('scroll', closeDropdownOnScroll);
> })
> ```

Можно в принципе и вынести
Просто это не переиспользуемый участок кода, и если выносить то только для того чтобы зрительно меньше кода внутри компоненты было

[the remark this answers is not in the document]
Comment (apequa-alan) [reply src/components/ExcelTable/ExcelTable.tsx]: @troynin Обновлено

File: src/components/ExcelTable/ExcelTable.tsx [hunk 2]
```diff
@@ -101,9 +102,13 @@ export const ExcelTable: React.FC<ExcelTableProps> = ({
   const gridRef = useRef<DataGridHandle>(null);
 
   useEffect(() => {
+    const exelTable = gridRef.current?.element;
```
Comment (lipgen) [inline src/components/ExcelTable/ExcelTable.tsx:105]: может excelTable? это специально так или опечатка?

**Review by lipgen — APPROVED** [review]

File: src/components/ExcelTable/ExcelTable.tsx (hunk 2 above)
Comment (apequa-alan) [reply src/components/ExcelTable/ExcelTable.tsx:105]: опечатка(

**Review by lipgen — APPROVED** [review]

**Pull request merged**